os/gtimer/gtimer_z_unit_entry_test.go

```diff
@@ -60,6 +60,27 @@ func TestJob_Singleton(t *testing.T) {
 	})
 }
 
+func TestJob_SingletonQuick(t *testing.T) {
+	gtest.C(t, func(t *gtest.T) {
+		timer := gtimer.New(gtimer.TimerOptions{
+			Quick: true,
+		})
+		array := garray.New(true)
+		job := timer.Add(ctx, 5*time.Second, func(ctx context.Context) {
+			array.Append(1)
+			time.Sleep(10 * time.Second)
+		})
+		t.Assert(job.IsSingleton(), false)
+		job.SetSingleton(true)
+		t.Assert(job.IsSingleton(), true)
+		time.Sleep(250 * time.Millisecond)
```

Comment:
If `Quick`, is it better no using sleep after adding but directly do the asserting?

Reply:
```

// loop starts the ticker using a standalone goroutine.
func (t *Timer) loop() {
	go func() {
		var (
			currentTimerTicks   int64
			timerIntervalTicker = time.NewTicker(t.options.Interval)
		)
		defer timerIntervalTicker.Stop()
		for {
			select {
			case <-timerIntervalTicker.C:
				// Check the timer status.
				switch t.status.Val() {
				case StatusRunning:
					// Timer proceeding.
					if currentTimerTicks = t.ticks.Add(1); currentTimerTicks >= t.queue.NextPriority() {
						t.proceed(currentTimerTicks)
					}

				case StatusStopped:
					// Do nothing.

				case StatusClosed:
					// Timer exits.
					return
				}
			}
		}
	}()
}

```

Because it is not really executed immediately, but as soon as possible, because it may need to wait for the delay time of `timerIntervalTicker`

并不是添加后就立即执行

而是需要等待添加后被轮询到后才会被执行

假设：
```
var timer = gtimer.New(gtimer.TimerOptions{
	Interval: 10000, //10秒
	Quick:    true,
})
```

那么它可能最多延迟10秒才会被执行，而默认设定的是100 即100ms

即最多延迟100ms